dogstatsd.py

```diff
@@ -318,18 +357,16 @@ def __init__(self, metrics_aggregator, host, port, forward_to_host=None, forward
                 log.exception("Error while setting up connection to external statsd server")
 
     def start(self):
-        """ Run the server. """
-        # Bind to the UDP socket.
-        # IPv4 only
-        self.socket = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
+        """
+        Run the server.
+        """
+        # Bind to the UDP socket in IPv4 and IPv6 compatibility mode
+        self.socket = socket.socket(socket.AF_INET6, socket.SOCK_DGRAM)
+        # Configure the socket so that it accepts connections from both
+        # IPv4 and IPv6 networks in a portable manner.
+        self.socket.setsockopt(IPPROTO_IPV6, IPV6_V6ONLY, 0)
         self.socket.setblocking(0)
-        try:
-            self.socket.bind(self.address)
-        except socket.gaierror:
-            if self.address[0] == 'localhost':
-                log.warning("Warning localhost seems undefined in your host file, using 127.0.0.1 instead")
-                self.address = ('127.0.0.1', self.address[1])
-                self.socket.bind(self.address)
+        self.socket.bind(self.address)
```

Comment:
for `gaierror` we are already covered b/c `self.address` handles that. We can add a more generic try/catch but what should we do in the exception handler?